ee/tables/jwt/table.go

```diff
@@ -0,0 +1,176 @@
+)
+
+type Table struct {
+	slogger *slog.Logger
+}
+
+func TablePlugin(slogger *slog.Logger) *table.Plugin {
+	columns := dataflattentable.Columns(
+		table.TextColumn("path"),
+		table.TextColumn("signature_key"),
+	)
+
+	t := &Table{
+		slogger: slogger.With("table", "kolide_jwt"),
+	}
+
+	return table.NewPlugin("kolide_jwt", columns, t.generate)
+}
+
+func (t *Table) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	paths := tablehelpers.GetConstraints(queryContext, "path")
+	if len(paths) < 1 {
+		return nil, fmt.Errorf("kolide_jwt requires at least one path to be specified")
+	}
+
+	var results []map[string]string
+	var keyMap map[string]string
+	keys := tablehelpers.GetConstraints(queryContext, "signature_key")
+
+	// This is perhaps a naive viewpoint, but if we want to return data even if we don't have a valid signature,
+	// then we don't entirely need to handle errors for unmarshaling passed-through keys.
+	for _, jsonString := range keys {
+		json.Unmarshal([]byte(jsonString), &keyMap)
+	}
+
+	for _, path := range paths {
+		for _, dataQuery := range tablehelpers.GetConstraints(queryContext, "query", tablehelpers.WithDefaults("*")) {
+			rawData, err := JWTRaw(path)
+			if err != nil {
+				t.slogger.Log(ctx, slog.LevelInfo, "error reading JWT data file", "err", err)
+				continue
+			}
+
+			row := map[string]interface{}{"verified": "UNKNOWN"}
+
+			// I've set 3 different states for if the signature is verified:
+			// VALID - The token parsed without errors and the signature was successfully validated.
+			// INVALID - The signature attempted to validate with the matched public key, but it was a bad key.
+			// UNKNOWN - The default state. This can mean that no key id matched, or simply no keys were provided to validate against.
+			token, err := jwt.ParseWithClaims(string(rawData), jwt.MapClaims{}, JWTKeyFunc(keyMap))
+			if err != nil {
+				t.slogger.Log(ctx, slog.LevelInfo, "error parsing token", "err", err)
+
+				if errors.Is(err, ErrParsingPemBlock) || errors.Is(err, ErrParsingPublicKey) {
+					row["verified"] = "INVALID"
+				}
+			} else {
+				row["verified"] = "VALID"
+			}
+
+			claims, ok := token.Claims.(jwt.MapClaims)
+			if !ok {
+				t.slogger.Log(ctx, slog.LevelInfo, "error parsing JWT claims")
+				continue
+			}
+
+			parsedClaims := map[string]interface{}{}
+			for k, v := range claims {
+				parsedClaims[k] = v
+			}
+
+			row["header"] = token.Header
+			row["claims"] = parsedClaims
+
+			flattenOpts := []dataflatten.FlattenOpts{
+				dataflatten.WithSlogger(t.slogger),
+				dataflatten.WithQuery(strings.Split(dataQuery, "/")),
+			}
+
+			flattened, err := dataflatten.Flatten(row, flattenOpts...)
+			if err != nil {
+				t.slogger.Log(ctx, slog.LevelInfo, "failure flattening JWT data", "err", err)
+				continue
+			}
+
+			rowData := map[string]string{
+				"path":          path,
+				"signature_key": strings.Join(keys, ""),
```

Comment:
I'm not sure this will work -- sqlite will require this column match whatever was provided. Having the join here feels wrong

Reply:
Yeah I need to revisit this perhaps. It does work, but I agree this feels dirty:

```
osquery> SELECT * FROM kolide_jwt WHERE key = 'verified' AND path IN ('/Users/micahsorenson/Downloads/data.zta', '/Users/micahsorenson/Downloads/data1.zta') AND signature_key IN ('', '{"US-2/v1":""}', '{"US-1/v1": "-----BEGIN PUBLIC KEY-----\nMIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEA1lq36T0TGIZet9K5KJf4\nzASFvcSCZkAZjenB4WLJsQOaKJzK44QieUMe868fYD8D4cPo+hbTI3Vozj2G8adu\n3pbKha9JJ8Ob4l+pR+zN9gyuJZ/ZIlm/S/ybkZBJDD2P9i2KHxHfg6en10jyZ40z\nyWu5Yaa8sTrSrURBs5FJxYDyvYY7w2RYR35mn5ZXZaAOFaPBwK2J8bNP0UFrTVr0\n5cyeN4plDy648XiXBB+BWD1PNTSX2o9nylXuVkWY/BWHvjlpunnE18gQOHJu6Mmk\nrbZtcRAVocYV+YIXqwOc2AwtkMLwl8OyI6cPuSy/v68gCv4Nt6EYJ4UJZvlNo++H\nMGsAe0kIUHurmawNd0A4s8wMYQkNSvJReiyuqTZeq+ZYfHmngWHmu0g2D3QimiDC\nQHDWj3yIaG6rlfrmAHeMTckqwv4AGzDq0QEMXmKXUuwXyU3gIV5hu0JyQgxsgybC\nDyZeoEYogHJUfwQZmVzr1T4mdI3irjN+0Y+rVD6mIVsCWu+jjtrHqGg/BfTuGnYc\nvktwtMFyWd9++FUGhriI+RzOqrhgJ0j8KArrzkIPky113mSdIZirmMfTeojGyVtm\nuVQmai7z3YEfCiddgBH5r4H1mJTPydXYzzpo8WqhxeMM592ludGwyOzo7Nv++0/Z\n/O2z/Q5uugXa9XMh946jd0cCAwEAAQ==\n-----END PUBLIC KEY-----\n", "US-2/v1": "-----BEGIN PUBLIC KEY-----\nMIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAuzYd6/byyZ3Oe63mBWTv\ns74onUzPdDz6LAbYwF63dARGn6vtTLll8Q39XxUkT5NyKEqjIbQcDDFE9xVOlrHI\nGWqBV9i2TAg1g0UdDwy1kyVxkwLrkPnVePknlMHuUs+eigHJ4O9NVotdnvOPSANU\nXzJR5repeXO47C0/FdcxYWwsI8UFk143wXZcRq7ZpMkDUBijQmfNGqP6CfFIoLsa\n6u0yMaWc+GYcK9PZsK6YIFAxgtn5UYz1Qb+HzBllXp8bOxNFi/JV2aOFziG1Ou71\n3DyRbsh7+a7J4FPSqil1UXY1E0+mPS8lHNxXcSVqYMQCGdluShYcxU7X9szZJIQq\nBqRZCQSbQb1JSD4j1wirUs+2+hikd4SXpvXzpT6XKdB2Zrk5L4+4EYhxP9GXXsiV\noNSDuaM4Di+bXL6LomhA+jVWPyEZ4JrU3htEb0aj0JOkLx7S18l0aU3W621ytnsN\nlF+BtaOE0UmpsUItORQPt7YbQm7MXS25TlAq595wJn3UzGupU2DYdJ7J4UwHDv3D\nlXD7pIgv786QKANxbS6B7lxlxJSBEMQE7BU3sjNIjjgAVXTYpFLxwzdL9Tk+DvNP\ntG5s+WornQEE0w5laxegMgkjpS20IM1ATQFIXDqt3r0jOU4jsOnbzUMZ/sH82FyG\nWYY+nGpXog0BeaL07JS9N1cCAwEAAQ==\n-----END PUBLIC KEY-----\n", "EU-1/v1": "-----BEGIN PUBLIC KEY-----\nMIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAqtnxZVNB4JddabltcdZ4\nh3MaqncLlT9+4HITJ45G51nv0rSl02DY1YkZCvVDdRVWEY1KLKwIEmXs1gigO8df\n6gwvVrixuLnXV8C3fzcOkK/f3CDZCT349rGhku708qkizUedwrgkGWOqk9WsM7wO\nTgQ9oYxEZl0Z3Bxs9Ht3E59sM691KaalVT87NPOk22nxZBng2EfcnLLmqvm1nAeS\n0fJ5ktoqjIyyTeial1LRZK96iWEZuUXZj72pzfg2kKVzLa0/5crlimXQVmW5cUqe\ni5u0c0qotLkXc2s/ait4y1SHF3+k4Q+ht4s+ziU1jgqoRx+oaY2pTNSsVGWHgenW\nSPeYKRRv3s5iTkdvl6xJ68Ip3EktB53Vw7OJWaIagqGD6Jv+5YMUUFNbt4e9alW1\nsHS1ZfKuphHk4RiH1lVcpJmzlM3XZyh1XtinxDwgH0d3kKyW7mPkmfsaKdgS8ui6\nj5WNqieJWg/Q5ZtBMxzm3wpNjYhgd352EV2BvUiCQ+EhpTw2uU7b9wrIPSM0Y25E\nU/jkIJq+o+gRAUtngM9/fkVDOGqKgzoYOGX7yezKN/6ONsrCAHOI3VZu7YSaAlJN\ncaDdLeevrZIw3oSaibGchXsk6PFJqVG6d7HSej7vv++O47PgwkLyepSHYT0SV+3q\nSWoRysQy6JfMFiy9sF4iKCMCAwEAAQ==\n-----END PUBLIC KEY-----\n"}') ORDER BY value;
{"time":"2024-05-31T18:46:12.95338Z","level":"INFO","msg":"error parsing token","launcher_run_id":"01HZ7ZAGA1YF05FXY4KG9GK455","table":"kolide_jwt","err":"token is unverifiable: error while executing keyfunc: no key id matched the JWT header key id"}
time=2024-05-31T18:46:12.953Z level=INFO source=/Users/micahsorenson/launcher/ee/tables/jwt/table.go:79 msg="error parsing token" launcher_run_id=01HZ7ZAGA1YF05FXY4KG9GK455 table=kolide_jwt err="token is unverifiable: error while executing keyfunc: no key id matched the JWT header key id"
{"time":"2024-05-31T18:46:12.961186Z","level":"INFO","msg":"error parsing token","launcher_run_id":"01HZ7ZAGA1YF05FXY4KG9GK455","table":"kolide_jwt","err":"token is unverifiable: error while executing keyfunc: error parsing PEM block containing the public key"}
time=2024-05-31T18:46:12.961Z level=INFO source=/Users/micahsorenson/launcher/ee/tables/jwt/table.go:79 msg="error parsing token" launcher_run_id=01HZ7ZAGA1YF05FXY4KG9GK455 table=kolide_jwt err="token is unverifiable: error while executing keyfunc: error parsing PEM block containing the public key"
{"time":"2024-05-31T18:46:12.962752Z","level":"INFO","msg":"error parsing token","launcher_run_id":"01HZ7ZAGA1YF05FXY4KG9GK455","table":"kolide_jwt","err":"token is unverifiable: error while executing keyfunc: no key id matched the JWT header key id"}
time=2024-05-31T18:46:12.962Z level=INFO source=/Users/micahsorenson/launcher/ee/tables/jwt/table.go:79 msg="error parsing token" launcher_run_id=01HZ7ZAGA1YF05FXY4KG9GK455 table=kolide_jwt err="token is unverifiable: error while executing keyfunc: no key id matched the JWT header key id"
{"time":"2024-05-31T18:46:12.967659Z","level":"INFO","msg":"error parsing token","launcher_run_id":"01HZ7ZAGA1YF05FXY4KG9GK455","table":"kolide_jwt","err":"token is unverifiable: error while executing keyfunc: error parsing PEM block containing the public key"}
time=2024-05-31T18:46:12.967Z level=INFO source=/Users/micahsorenson/launcher/ee/tables/jwt/table.go:79 msg="error parsing token" launcher_run_id=01HZ7ZAGA1YF05FXY4KG9GK455 table=kolide_jwt err="token is unverifiable: error while executing keyfunc: error parsing PEM block containing the public key"
      fullkey = verified
       parent =
          key = verified
        value = INVALID
        query = *
         path = /Users/micahsorenson/Downloads/data.zta
signature_key = {"US-2/v1":""}

      fullkey = verified
       parent =
          key = verified
        value = INVALID
        query = *
         path = /Users/micahsorenson/Downloads/data1.zta
signature_key = {"US-2/v1":""}

      fullkey = verified
       parent =
          key = verified
        value = UNKNOWN
        query = *
         path = /Users/micahsorenson/Downloads/data.zta
signature_key =

      fullkey = verified
       parent =
          key = verified
        value = UNKNOWN
        query = *
         path = /Users/micahsorenson/Downloads/data1.zta
signature_key =

      fullkey = verified
       parent =
          key = verified
        value = VALID
        query = *
         path = /Users/micahsorenson/Downloads/data.zta
signature_key = {"US-1/v1": "-----BEGIN PUBLIC KEY-----\nMIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEA1lq36T0TGIZet9K5KJf4\nzASFvcSCZkAZjenB4WLJsQOaKJzK44QieUMe868fYD8D4cPo+hbTI3Vozj2G8adu\n3pbKha9JJ8Ob4l+pR+zN9gyuJZ/ZIlm/S/ybkZBJDD2P9i2KHxHfg6en10jyZ40z\nyWu5Yaa8sTrSrURBs5FJxYDyvYY7w2RYR35mn5ZXZaAOFaPBwK2J8bNP0UFrTVr0\n5cyeN4plDy648XiXBB+BWD1PNTSX2o9nylXuVkWY/BWHvjlpunnE18gQOHJu6Mmk\nrbZtcRAVocYV+YIXqwOc2AwtkMLwl8OyI6cPuSy/v68gCv4Nt6EYJ4UJZvlNo++H\nMGsAe0kIUHurmawNd0A4s8wMYQkNSvJReiyuqTZeq+ZYfHmngWHmu0g2D3QimiDC\nQHDWj3yIaG6rlfrmAHeMTckqwv4AGzDq0QEMXmKXUuwXyU3gIV5hu0JyQgxsgybC\nDyZeoEYogHJUfwQZmVzr1T4mdI3irjN+0Y+rVD6mIVsCWu+jjtrHqGg/BfTuGnYc\nvktwtMFyWd9++FUGhriI+RzOqrhgJ0j8KArrzkIPky113mSdIZirmMfTeojGyVtm\nuVQmai7z3YEfCiddgBH5r4H1mJTPydXYzzpo8WqhxeMM592ludGwyOzo7Nv++0/Z\n/O2z/Q5uugXa9XMh946jd0cCAwEAAQ==\n-----END PUBLIC KEY-----\n", "US-2/v1": "-----BEGIN PUBLIC KEY-----\nMIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAuzYd6/byyZ3Oe63mBWTv\ns74onUzPdDz6LAbYwF63dARGn6vtTLll8Q39XxUkT5NyKEqjIbQcDDFE9xVOlrHI\nGWqBV9i2TAg1g0UdDwy1kyVxkwLrkPnVePknlMHuUs+eigHJ4O9NVotdnvOPSANU\nXzJR5repeXO47C0/FdcxYWwsI8UFk143wXZcRq7ZpMkDUBijQmfNGqP6CfFIoLsa\n6u0yMaWc+GYcK9PZsK6YIFAxgtn5UYz1Qb+HzBllXp8bOxNFi/JV2aOFziG1Ou71\n3DyRbsh7+a7J4FPSqil1UXY1E0+mPS8lHNxXcSVqYMQCGdluShYcxU7X9szZJIQq\nBqRZCQSbQb1JSD4j1wirUs+2+hikd4SXpvXzpT6XKdB2Zrk5L4+4EYhxP9GXXsiV\noNSDuaM4Di+bXL6LomhA+jVWPyEZ4JrU3htEb0aj0JOkLx7S18l0aU3W621ytnsN\nlF+BtaOE0UmpsUItORQPt7YbQm7MXS25TlAq595wJn3UzGupU2DYdJ7J4UwHDv3D\nlXD7pIgv786QKANxbS6B7lxlxJSBEMQE7BU3sjNIjjgAVXTYpFLxwzdL9Tk+DvNP\ntG5s+WornQEE0w5laxegMgkjpS20IM1ATQFIXDqt3r0jOU4jsOnbzUMZ/sH82FyG\nWYY+nGpXog0BeaL07JS9N1cCAwEAAQ==\n-----END PUBLIC KEY-----\n", "EU-1/v1": "-----BEGIN PUBLIC KEY-----\nMIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAqtnxZVNB4JddabltcdZ4\nh3MaqncLlT9+4HITJ45G51nv0rSl02DY1YkZCvVDdRVWEY1KLKwIEmXs1gigO8df\n6gwvVrixuLnXV8C3fzcOkK/f3CDZCT349rGhku708qkizUedwrgkGWOqk9WsM7wO\nTgQ9oYxEZl0Z3Bxs9Ht3E59sM691KaalVT87NPOk22nxZBng2EfcnLLmqvm1nAeS\n0fJ5ktoqjIyyTeial1LRZK96iWEZuUXZj72pzfg2kKVzLa0/5crlimXQVmW5cUqe\ni5u0c0qotLkXc2s/ait4y1SHF3+k4Q+ht4s+ziU1jgqoRx+oaY2pTNSsVGWHgenW\nSPeYKRRv3s5iTkdvl6xJ68Ip3EktB53Vw7OJWaIagqGD6Jv+5YMUUFNbt4e9alW1\nsHS1ZfKuphHk4RiH1lVcpJmzlM3XZyh1XtinxDwgH0d3kKyW7mPkmfsaKdgS8ui6\nj5WNqieJWg/Q5ZtBMxzm3wpNjYhgd352EV2BvUiCQ+EhpTw2uU7b9wrIPSM0Y25E\nU/jkIJq+o+gRAUtngM9/fkVDOGqKgzoYOGX7yezKN/6ONsrCAHOI3VZu7YSaAlJN\ncaDdLeevrZIw3oSaibGchXsk6PFJqVG6d7HSej7vv++O47PgwkLyepSHYT0SV+3q\nSWoRysQy6JfMFiy9sF4iKCMCAwEAAQ==\n-----END PUBLIC KEY-----\n"}

      fullkey = verified
       parent =
          key = verified
        value = VALID
        query = *
         path = /Users/micahsorenson/Downloads/data1.zta
signature_key = {"US-1/v1": "-----BEGIN PUBLIC KEY-----\nMIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEA1lq36T0TGIZet9K5KJf4\nzASFvcSCZkAZjenB4WLJsQOaKJzK44QieUMe868fYD8D4cPo+hbTI3Vozj2G8adu\n3pbKha9JJ8Ob4l+pR+zN9gyuJZ/ZIlm/S/ybkZBJDD2P9i2KHxHfg6en10jyZ40z\nyWu5Yaa8sTrSrURBs5FJxYDyvYY7w2RYR35mn5ZXZaAOFaPBwK2J8bNP0UFrTVr0\n5cyeN4plDy648XiXBB+BWD1PNTSX2o9nylXuVkWY/BWHvjlpunnE18gQOHJu6Mmk\nrbZtcRAVocYV+YIXqwOc2AwtkMLwl8OyI6cPuSy/v68gCv4Nt6EYJ4UJZvlNo++H\nMGsAe0kIUHurmawNd0A4s8wMYQkNSvJReiyuqTZeq+ZYfHmngWHmu0g2D3QimiDC\nQHDWj3yIaG6rlfrmAHeMTckqwv4AGzDq0QEMXmKXUuwXyU3gIV5hu0JyQgxsgybC\nDyZeoEYogHJUfwQZmVzr1T4mdI3irjN+0Y+rVD6mIVsCWu+jjtrHqGg/BfTuGnYc\nvktwtMFyWd9++FUGhriI+RzOqrhgJ0j8KArrzkIPky113mSdIZirmMfTeojGyVtm\nuVQmai7z3YEfCiddgBH5r4H1mJTPydXYzzpo8WqhxeMM592ludGwyOzo7Nv++0/Z\n/O2z/Q5uugXa9XMh946jd0cCAwEAAQ==\n-----END PUBLIC KEY-----\n", "US-2/v1": "-----BEGIN PUBLIC KEY-----\nMIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAuzYd6/byyZ3Oe63mBWTv\ns74onUzPdDz6LAbYwF63dARGn6vtTLll8Q39XxUkT5NyKEqjIbQcDDFE9xVOlrHI\nGWqBV9i2TAg1g0UdDwy1kyVxkwLrkPnVePknlMHuUs+eigHJ4O9NVotdnvOPSANU\nXzJR5repeXO47C0/FdcxYWwsI8UFk143wXZcRq7ZpMkDUBijQmfNGqP6CfFIoLsa\n6u0yMaWc+GYcK9PZsK6YIFAxgtn5UYz1Qb+HzBllXp8bOxNFi/JV2aOFziG1Ou71\n3DyRbsh7+a7J4FPSqil1UXY1E0+mPS8lHNxXcSVqYMQCGdluShYcxU7X9szZJIQq\nBqRZCQSbQb1JSD4j1wirUs+2+hikd4SXpvXzpT6XKdB2Zrk5L4+4EYhxP9GXXsiV\noNSDuaM4Di+bXL6LomhA+jVWPyEZ4JrU3htEb0aj0JOkLx7S18l0aU3W621ytnsN\nlF+BtaOE0UmpsUItORQPt7YbQm7MXS25TlAq595wJn3UzGupU2DYdJ7J4UwHDv3D\nlXD7pIgv786QKANxbS6B7lxlxJSBEMQE7BU3sjNIjjgAVXTYpFLxwzdL9Tk+DvNP\ntG5s+WornQEE0w5laxegMgkjpS20IM1ATQFIXDqt3r0jOU4jsOnbzUMZ/sH82FyG\nWYY+nGpXog0BeaL07JS9N1cCAwEAAQ==\n-----END PUBLIC KEY-----\n", "EU-1/v1": "-----BEGIN PUBLIC KEY-----\nMIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAqtnxZVNB4JddabltcdZ4\nh3MaqncLlT9+4HITJ45G51nv0rSl02DY1YkZCvVDdRVWEY1KLKwIEmXs1gigO8df\n6gwvVrixuLnXV8C3fzcOkK/f3CDZCT349rGhku708qkizUedwrgkGWOqk9WsM7wO\nTgQ9oYxEZl0Z3Bxs9Ht3E59sM691KaalVT87NPOk22nxZBng2EfcnLLmqvm1nAeS\n0fJ5ktoqjIyyTeial1LRZK96iWEZuUXZj72pzfg2kKVzLa0/5crlimXQVmW5cUqe\ni5u0c0qotLkXc2s/ait4y1SHF3+k4Q+ht4s+ziU1jgqoRx+oaY2pTNSsVGWHgenW\nSPeYKRRv3s5iTkdvl6xJ68Ip3EktB53Vw7OJWaIagqGD6Jv+5YMUUFNbt4e9alW1\nsHS1ZfKuphHk4RiH1lVcpJmzlM3XZyh1XtinxDwgH0d3kKyW7mPkmfsaKdgS8ui6\nj5WNqieJWg/Q5ZtBMxzm3wpNjYhgd352EV2BvUiCQ+EhpTw2uU7b9wrIPSM0Y25E\nU/jkIJq+o+gRAUtngM9/fkVDOGqKgzoYOGX7yezKN/6ONsrCAHOI3VZu7YSaAlJN\ncaDdLeevrZIw3oSaibGchXsk6PFJqVG6d7HSej7vv++O47PgwkLyepSHYT0SV+3q\nSWoRysQy6JfMFiy9sF4iKCMCAwEAAQ==\n-----END PUBLIC KEY-----\n"}
osquery>
```

Reply:
Haha yeah why the hell did I write it like this 🤦 the whole point of passing in a hash was to handle this lmao. My brain on 60 hours of no sleep should probably not be coding.

Reply:
Your example SQL sends in N different possible signature keys, right? Which sometimes ends up invoking generate N times, so we'll potentially return a bunch of rows, or log errors. I suspect we should pass in a single structure, maybe base64 encoded to make it a little nicer to work with? Not sure.

Reply:
TBH the only reason we need to return it, is so that sqlite doesn't filter it out for failing the where clause. And sadly, the sdk doesn't support `HIDDEN` columns

Reply:
Yeah my example SQL was me just enumerating it, but the use should be passing in a single JSON string with the keys. Also I didn't think that the only reason it was working out is because osquery doesn't pass IN constraints all at once. Which is exactly the thing I have a PR open to change, so that would've broken this if it changed.

Reply:
I'm open to going about this in a different way. Admittedly my brain was not very functional when I wrote this.